include/bluetooth/gatt.h

```diff
@@ -1018,6 +1021,8 @@ struct bt_gatt_indicate_params {
 	const struct bt_gatt_attr *attr;
 	/** Indicate Value callback */
 	bt_gatt_indicate_func_t func;
+	/** Host populated counter, this struct can be freed when equal to 0 */
+	int ref;
```

Comment:
On second thought, let's be consistent with an already existing mechanism for this, the L2CAP destroy callback: 
https://github.com/zephyrproject-rtos/zephyr/blob/master/include/bluetooth/l2cap.h#L52

Perhaps `typedef void (*bt_gatt_indicate_params_destroy_t)(struct bt_gatt_indicate_params *params);`

One way to go about this would be to add the callback and then manage the reference count internally and expose only a `_ref` or `void *private` member in this structure.

Reply:
Where would this callback be added? To the `struct bt_gatt_indicate_params`?
If so, is this the accepted path forward?
My argument against it would be that we are adding another 4 bytes to the structure, when a usable callback is already run in the right places (I don't think this is this is the case for l2cap channels, but I'm not familiar with them).
The internal code then looks like.
```
    params->func(conn, params, err);
    params->_ref--;
    if (params->destroy && (params->_ref == 0)) {
        params->destroy(params);
    }
```

Reply:
@JordanYates Yes, that would the accepted path forward.
To prevent increasing the size of the structure the _ref could be a uint16_t and placed at the end after len. The ref count would be maximum one per connection, so that should be enough. Could even be uint8_t I think.